libs/grok/src/main/java/org/elasticsearch/grok/Grok.java

```diff
@@ -220,8 +241,20 @@ public boolean match(String text) {
         byte[] textAsBytes = text.getBytes(StandardCharsets.UTF_8);
         Map<String, Object> fields = new HashMap<>();
         Matcher matcher = compiledExpression.matcher(textAsBytes);
-        int result = matcher.search(0, textAsBytes.length, Option.DEFAULT);
-        if (result != -1 && compiledExpression.numberOfNames() > 0) {
+        int result;
+        try {
+            threadWatchdog.register();
+            result = matcher.search(0, textAsBytes.length, Option.DEFAULT);
+        } finally {
+            threadWatchdog.unregister();
+        }
+        if (result == Matcher.INTERRUPTED) {
+            throw new IllegalArgumentException("grok pattern matching was interrupted after [" +
+                threadWatchdog.maxExecutionTime() + "] ms");
```

Comment:
I think that if we keep `ThreadWatchdog#maxExecutionTime` as `TimeValue` then we can let `TimeValue` do the formatting for us?